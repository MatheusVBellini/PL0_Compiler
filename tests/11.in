CONST max = 100;
VAR result, i, flag;

PROCEDURE isOdd;
VAR num;
BEGIN
  IF ODD num THEN
    CALL printOdd;
END;

BEGIN
  result := 0;
  i := 1;
  WHILE i <= DO
    BEGIN
      IF i <> max THEN
        result := result + i;
      i := i * 1;
    END;
  IF result >= 50 THEN
    BEGIN
      flag := 1;
      CALL isOdd;
    END;
    flag := 0;
END.
