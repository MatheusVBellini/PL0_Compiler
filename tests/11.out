CONST, CONST
max, ident
=, simbolo_igual
100, numero
;, simbolo_ponto_virgula
VAR, VAR
result, ident
,, simbolo_virgula
i, ident
,, simbolo_virgula
flag, ident
;, simbolo_ponto_virgula
PROCEDURE, PROCEDURE
isOdd, ident
;, simbolo_ponto_virgula
VAR, VAR
num, ident
;, simbolo_ponto_virgula
BEGIN, BEGIN
IF, IF
ODD, ODD
num, ident
THEN, THEN
CALL, CALL
printOdd, ident
;, simbolo_ponto_virgula
END, END
;, simbolo_ponto_virgula
BEGIN, BEGIN
result, ident
:=, simbolo_atribuicao
0, numero
;, simbolo_ponto_virgula
i, ident
:=, simbolo_atribuicao
1, numero
;, simbolo_ponto_virgula
WHILE, WHILE
i, ident
<=, simbolo_menor_igual
max, ident
DO, DO
BEGIN, BEGIN
IF, IF
i, ident
<>, simbolo_diferente
max, ident
THEN, THEN
result, ident
:=, simbolo_atribuicao
result, ident
+, simbolo_mais
i, ident
;, simbolo_ponto_virgula
i, ident
:=, simbolo_atribuicao
i, ident
*, simbolo_vezes
1, numero
;, simbolo_ponto_virgula
END, END
;, simbolo_ponto_virgula
IF, IF
result, ident
>=, simbolo_maior_igual
50, numero
THEN, THEN
BEGIN, BEGIN
flag, ident
:=, simbolo_atribuicao
1, numero
;, simbolo_ponto_virgula
CALL, CALL
isOdd, ident
;, simbolo_ponto_virgula
END, END
flag, ident
:=, simbolo_atribuicao
0, numero
;, simbolo_ponto_virgula
END, END
., simbolo_ponto
